<!-- -*- mode: XML -*- -->
<launch>
  <master auto="start"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(find renpsm)maps/rrt1.yaml" />
  <node pkg="amcl" type="amcl" name="amcl" args="scan:=/rosaria/sim_lms1xx_1_laserscan">
    <param name="use_map_topic" value="true" />
    <param name="odom_model_type" value="diff"/>
    <param name="transform_tolerance" value="0.9" /> <!--"0.2" />-->
    <param name="gui_publish_rate" value="10.0"/>
    <param name="laser_max_beams" value="50"/>
    <param name="min_particles" value="200"/> <!--200-->
    <param name="max_particles" value="1000"/> <!--1000-->
    <param name="kld_err" value="0.05"/>
    <param name="kld_z" value="0.99"/>
    <param name="odom_alpha1" value="0.2"/><!-- 0.1 def:0.2 -->
    <param name="odom_alpha2" value="0.1"/><!-- 0.1 def:0.2 -->
    <!-- translation std dev, m -->
    <param name="odom_alpha3" value="0.2"/><!--0.8 def:0.2 -->
    <param name="odom_alpha4" value="0.1"/><!--def:0.2 -->
    <param name="odom_alpha5" value="0.0"/>
    <param name="laser_max_range" value="10"/> <!-- new -->
    <param name="laser_min_range" value="0.5"/> <!-- new -->
    <param name="laser_z_hit" value="0.85"/> <!-- 0.5 -->
    <param name="laser_z_short" value="0.05"/>
    <param name="laser_z_max" value="0.05"/>
    <param name="laser_z_rand" value="0.05"/> <!-- 0.5 -->
    <param name="laser_sigma_hit" value="0.05"/> <!-- 0.05  default:0.2-->
    <param name="laser_lambda_short" value="0.1"/>
    <param name="laser_model_type" value="likelihood_field"/>
    <!-- <param name="laser_model_type" value="beam"/> -->
    <param name="laser_likelihood_max_dist" value="2.0"/>
    <param name="update_min_d" value="0.3"/>
    <param name="update_min_a" value="0.2"/>
    <param name="odom_frame_id" value="/odom"/>
    <param name="base_frame_id" value="/base_link"/>
    <param name="global_frame_id" value="/map"/>
    <param name="resample_interval" value="2"/>
    <param name="recovery_alpha_slow" value="0"/> <!--0.001--><!-- 0 -->
    <param name="recovery_alpha_fast" value="0"/> <!--0.1--><!-- 0 -->
    <param name="initial_pose_x" value="8.52" /> <!--11.1-->
    <param name="initial_pose_y" value="7.5" /> <!--5.2-->
    <param name="initial_pose_a" value="0" /><!-- -2.79 radians -->
</node>
  <arg name="urdf" default="$(find amr_robots_description)/urdf/pioneer3dx.urdf" />
  <arg name="joint_state_gui" default="false" />
  <include file="$(find amr_robots_launchfiles)/rosaria.launch">
    <arg name="urdf" value="$(arg urdf)"/>
    <!-- arg name="joint_state_gui" value="$(arg joint_state_gui)" / -->
  </include>
  <node name="renpsm" pkg="renpsm" type="renpsm_node" output="screen" >
	<param name="goal_topic" value="/move_base_simple/goal"/>
	<param name="odom_topic" value="/rosaria/pose"/>
  </node>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find amr_robots_launchfiles)/rosaria.rviz" required="true" />
</launch>
